File: test/fixtures/csharp/patch.cs
using System.Net.Http;
using System.Net.Http.Headers;

HttpClient client = new HttpClient();

HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Patch, "http://localhost:28139/go/api/agents/adb9540a-b954-4571-9d9b-2f330739d4da");

request.Headers.Add("Accept", "application/vnd.go.cd.v4+json");
request.Headers.Add("Authorization", "Basic " + Convert.ToBase64String(System.Text.ASCIIEncoding.ASCII.GetBytes("username:password")));

request.Content = new StringContent("{\n        \"hostname\": \"agent02.example.com\",\n        \"agent_config_state\": \"Enabled\",\n        \"resources\": [\"Java\",\"Linux\"],\n        \"environments\": [\"Dev\"]\n        }");
request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");

HttpResponseMessage response = await client.SendAsync(request);
response.EnsureSuccessStatusCode();
string responseBody = await response.Content.ReadAsStringAsync();

File: test/fixtures/clojure/patch.clj
(require '[cheshire.core :as json])
(require '[clj-http.client :as client])

(client/patch "http://localhost:28139/go/api/agents/adb9540a-b954-4571-9d9b-2f330739d4da"
              {:headers {"Accept" "application/vnd.go.cd.v4+json"}
               :basic-auth ["username" "password"]
               :form-params {:hostname "agent02.example.com"
                             :agent_config_state "Enabled"
                             :resources ["Java" "Linux"]
                             :environments ["Dev"]}
               ;; :body "{\n        \"hostname\": \"agent02.example.com\",\n        \"agent_config_state\": \"Enabled\",\n        \"resources\": [\"Java\",\"Linux\"],\n        \"environments\": [\"Dev\"]\n        }"
               :content-type :json})
